internal/db/secrets.go

```diff
@@ -0,0 +1,262 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/internal/db/dbconn"
+)
+
+// This object provides access to the secrets table
+type secrets struct{}
+
+// Returned when we can't retrieve the specific crypt object that we need
+type SecretNotFoundError struct {
+}
+
+// We always try to find *one* object.
+func (err *SecretNotFoundError) Error() string {
+	return "failed to find matching secret"
+}
+
+// Delete by object id
+func (s *secrets) Delete(ctx context.Context, id int32) error {
+	if Mocks.Secrets.Delete != nil {
+		return Mocks.Secrets.Delete(ctx, id)
+	}
+
+	sqlQ := sqlf.Sprintf(
+		`DELETE FROM
+			secrets
+		WHERE
+			id=$1
+		`, id)
```

Comment:
It's probably overkill to make it multi-line :) but up to you!
```suggestion
	q := sqlf.Sprintf(`DELETE FROM secrets WHERE id=%s`, id)
```

Variable name `q` is what we use most often for the query object. Besides, using `$1` actually won't work, `sqlf` package requires using `%s` (and the package will replace these placeholders with correct bind vars in `q.Query(sqlf.PostgresBindVar)`).

Reply:
I accepted the rest. Currently I'm on a multi-line SQL kick thanks to grafana - I find it more readable.